### DISCUSSION TITLE
Is USER_SETUP_ID available at compile time?

### DISCUSSION THREAD

**Discussion by elaw01887** [Bodmer/TFT_eSPI#3549] [Q&A - General] [answered]
As the title says I'm wondering if USER_SETUP_ID is available at compile time (in the Arduino IDE)?

If I do Serial.println(USER_SETUP_ID);, I get the output I expect.

But if I do this:

```
#if (USER_SETUP_ID != 0x199801)
#error Wrong USER_SETUP_ID from TFT_eSPI - do you have the correct user setup selected?
#endif

```
... it doesn't seem to work.

**Comment by Basitadam** [comment, marked as answer]
Hi @elaw01887,

I think depends where you put the statement. Example:
My USER_SETUP_ID is 70,  does not work if I put here, before the library include line:

![image](https://github.com/user-attachments/assets/c773516b-5f6f-47e6-ac7e-d782bb961374)



Works when I put after:

![image](https://github.com/user-attachments/assets/be5a5eee-1966-44fb-9e01-3929fcb9053e)

**Comment by elaw01887** [comment]
Thanks... that fixed my problem!

**Reply by Basitadam** [reply]
```
HappyFace();
```
